hyperspy/io_plugins/fei.py

```diff
@@ -316,6 +317,115 @@ def get_xml_info_from_emi(emi_file):
     return objects[:-1]
 
 
+def get_calibration_from_position(position):
+    """Compute the size, scale and offset of a linear axis from coordinates.
+
+    This function assumes rastering on a regular grid for the full size of
+    each dimension before rastering over another one. Fox example: a11, a12,
+    a13, a21, a22, a23 for a 2x3 grid.
+
+    Parameters
+    ----------
+    position: numpy array.
+        Position coordinates of the axis. Normally as in PositionX/Y of the
+        ser file.
+
+    Returns
+    -------
+    axis_attr: dictionary with `size`, `scale`, `offeset` keys.
+
+    """
+    offset = position[0]
+    for i, x in enumerate(position):
+        if x != position[0]:
+            break
+    if i == len(position) - 1:
+        # No scanning over this axis
+        scale = 0
+        size = 0
+    else:
+        scale = x - position[0]
+        if i == 1:  # Rastering over this dimension first
+            for j, x in enumerate(position[1:]):
+                if x == position[0]:
+                    break
+            size = j + 1
+        else:  # Second rastering dimension
+            size = len(position) / i
+    axis_attr = {"size": size, "scale": scale, "offset": offset}
+    return axis_attr
+
+
+def get_axes_from_position(header, data):
+    array_shape = []
+    axes = []
+    array_size = int(header["ValidNumberElements"])
+    if data["TagTypeID"][0] == XY_TAG_ID:
+        xcal = get_calibration_from_position(data["PositionX"])
+        ycal = get_calibration_from_position(data["PositionY"])
+        if xcal["size"] == 0 and ycal["size"] != 0:
+            # Vertical line scan
+            axes.append({
+                'name': "x",
+                'units': "meters",
+                'index_in_array': 0,
+            })
+            axes[-1].update(xcal)
+            array_shape.append(axes[-1]["size"])
+
+        elif xcal["size"] != 0 and ycal["size"] == 0:
+            # Horizontal line scan
+            axes.append({
+                'name': "y",
+                'units': "meters",
+                'index_in_array': 0,
+            })
+            axes[-1].update(ycal)
+            array_shape.append(axes[-1]["size"])
+
+        elif xcal["size"] * ycal["size"] == array_size:
+            # Image
+            axes.append({
+                'name': "y",
+                'units': "meters",
+                'index_in_array': 0,
+            })
+            axes[-1].update(ycal)
+            array_shape.append(axes[-1]["size"])
+            axes.append({
+                'name': "x",
+                'units': "meters",
+                'index_in_array': 1,
+            })
+            axes[-1].update(xcal)
+            array_shape.append(axes[-1]["size"])
+        elif xcal["size"] == ycal["size"] == array_size:
+            # Oblique line scan
```

Comment:
I haven't actually tested this part of the code because I don't have any oblique line scan recorded like this among my ser files. However, I think that it should be correct. An oblique line scan should look like a sawtooth with just one tooth, regardless of the image being rectangular of square. For the same reason the size should be len(PositionX) == len(PositionY) == array_size